builds/install/misc/firebird.conf

```diff
@@ -244,11 +244,17 @@
 # ----------------------------
 # Number of cached database pages
 #
-# This sets the number of pages from any one database that can be held
-# in cache at once. If you increase this value, the engine will
-# allocate more pages to the cache for every database. By default, the
-# SuperServer allocates 2048 pages for each database and the classic
-# allocates 256 pages per client connection per database.
+# This setting is used to cache in memory database pages. 
+# It specifies maximum number of pages which can be cached.
+# In SuperServer it is per database, in Classic per client connection.
+# Per-database configurable. This value can be overrided by databases.conf or 
+# directly in database file by issuing command:
+# gfix -b[uffers] BUFFERS
+# Priority of settings is: directly in database, databases.conf, firebird.conf.
+# The SuperServer default value is 2048 pages for each database and the classic
+# value is 256 pages per client connection per database.
+# As this value means maximum number of pages cached, at start server use minimal RAM.
```

Comment:
Engine allocates all buffers at the attachment. It's true both for SuperServer and for ClassicServer. Therefore, SuperServer allocates all buffers at the first connection. ClassicServer allocates buffers for every attachment and this number depends on number of attachments.
I think suggested description is not complete as well and doesn't sole original question.

Reply:
@romansimakov 
I did not observe such thing like SuperServer allocation of all buffers. I start server, connect to database and server still use low RAM. Only if i select from multiple tables it increase RAM usage.

Reply:
> @romansimakov I did not observe such thing like SuperServer allocation of all buffers. I start server, connect to database and server still use low RAM. Only if i select from multiple tables it increase RAM usage. But i still use FB3. Maybe was some changes in recent version of FB?

What's your OS and how do you see? I'm afraid OS just doesn't allocate asked memory phisically and leaves it as virtually allocated. When engine addresses a memory page maps on phisical memory and from that time OS account it as used by the process. 

For example, on my Ubuntu I can see the following for 2048 cache on SuperServer:
1) Just after server start
```
roman:bin$ grep Vm /proc/`pidof rdbserver`/status 
VmPeak:	  470764 kB
VmSize:	  405680 kB
VmHWM:	   42504 kB
VmRSS:	   42504 kB
VmData:	   58240 kB
```
2) After connect to a big (80GB) database:
roman:bin$ grep Vm /proc/`pidof rdbserver`/status 
```
VmPeak:	 1384600 kB
VmSize:	 1319452 kB
VmHWM:	   72432 kB
VmRSS:	   72432 kB
VmData:	  632884 kB
```
3) After select from a big table:
roman:bin$ grep Vm /proc/`pidof rdbserver`/status 
```
VmPeak:	 1384600 kB
VmSize:	 1319068 kB
VmHWM:	  103860 kB
VmRSS:	  103860 kB
VmData:	  632500 kB
```
4) After disconnect
roman:bin$ grep Vm /proc/`pidof rdbserver`/status 
```
VmPeak:	 1384600 kB
VmSize:	  764724 kB
VmHWM:	  103860 kB
VmRSS:	   40868 kB
VmData:	   90628 kB
```
Note how VmRSS and VmSize change.

Reply:
I.e. it's controlled not by the Engine but by OS

Reply:
I am working only on Windows, and i simply looked into TaskManager->Resource monitor
At start FB (SuperServer) show really low RAM usage, only when i do select from multiple tables it increase:
![image](https://github.com/FirebirdSQL/firebird/assets/14888294/28ad49af-2fed-4a11-8721-f50faca4efe3)

Reply:
Get a difference between virtual memory (working set) and physical (committed) memory. When you allocate memory, virtual memory counter is increased. When you access that allocated memory, physical memory counter is increased. So Firebird allocates (read: reserves in virtual memory) the whole page cache at startup, but its usage (physical/committed memory) will grow only when pages are read into the cache.

Reply:
I suppose that talking about Phisical Memory usage is more interesting for people which read config values description and what they observe after FB start. But my text was only proposition of description, fill free to extend that info of Virtual Memory preservation and any other info which help to understand that parameter.